concepts/patterns/about.md

```diff
@@ -0,0 +1,108 @@
+# Introduction
+
+Recall from the Fundamentals concept, an AWK program is composed of **pattern-action pairs**.
+
+```awk
+pattern1 { action1 }
+pattern2 { action2 }
+...
+```
+
+## What do we mean by "pattern"?
+
+The "pattern" is any AWK expression.
+The truthiness of the expression's result determines if the action is executed.
+
+## The Empty Pattern
+
+The pattern can be omitted.
+In this case, the action is performed for every record.
+
+We can print all the usernames in the passwd file.
+
+```sh
+awk -F: '{print $1}' /etc/passwd
+```
+
+## Regular Expressions
+
+AWK can compare strings against regular expressions to obtain a boolean result.
+
+Use the `~` regex-matching operator to match a particular field. 
+This operator takes a string as the left-hand operand and a regular expression as the right-hand operand.
+A regular expression literal is enclosed in `/` slashes.
+
+To find the user in the passed file who log in with bash:
+
+```sh
+awk -F: '$7 ~ /bash/ {print $1}' /etc/passwd
+```
+
+`!~` is the "regex does **not** match" operator.
+
+If you want to match a regex against the whole current record, there's a shorthand syntax for it:
+the `$0 ~ /regex` expression can be simply expressed as:
```

Comment:
```suggestion
To match a regex against the whole current record, you can do `$0 ~ /regex/`.
As a shorthand for this, you can include only the regex part and this will check the whole current record.
```